be/src/service/internal_service.cpp

```diff
@@ -188,18 +205,76 @@ Status PInternalServiceImplBase<T>::_exec_plan_fragment(brpc::Controller* cntl)
               << ", coord=" << t_request.coord << ", backend=" << t_request.backend_num
               << ", is_pipeline=" << is_pipeline << ", chunk_size=" << t_request.query_options.batch_size;
     if (is_pipeline) {
-        auto fragment_executor = std::make_unique<starrocks::pipeline::FragmentExecutor>();
-        auto status = fragment_executor->prepare(_exec_env, t_request);
-        if (status.ok()) {
-            return fragment_executor->execute(_exec_env);
-        } else {
-            return status.is_duplicate_rpc_invocation() ? Status::OK() : status;
+        return _exec_plan_fragment_by_pipeline(t_request, t_request);
+    } else {
+        return _exec_plan_fragment_by_non_pipeline(t_request);
+    }
+}
+
+template <typename T>
+Status PInternalServiceImplBase<T>::_exec_batch_plan_fragments(brpc::Controller* cntl) {
+    auto ser_request = cntl->request_attachment().to_string();
+    TExecBatchPlanFragmentsParams t_batch_requests;
+    {
+        const uint8_t* buf = (const uint8_t*)ser_request.data();
+        uint32_t len = ser_request.size();
+        RETURN_IF_ERROR(deserialize_thrift_msg(buf, &len, TProtocolType::BINARY, &t_batch_requests));
+    }
+
+    auto& common_request = t_batch_requests.common_param;
+    auto& unique_requests = t_batch_requests.unique_param_per_instance;
+
+    if (unique_requests.empty()) {
+        return Status::OK();
+    }
+    RETURN_IF_ERROR(_exec_plan_fragment_by_pipeline(common_request, unique_requests[0]));
```

Comment:
the optimization only works in pipeline engine?

Reply:
Yes, this PR is just for pipeline engine.